### PR TITLE
Changing Compare methods to compare current value to other in "this" …

### PR DESCRIPTION
**Pull request by tmilnthorp**
Changing Compare methods to compare current value to other in "this" units. This avoids scaling issues (and converting two quantities when we only need to potentially convert one) when both "this" and "other" are not defined in base units. "This" is the reference value.

### CODE REVIEW

File: UnitsNet/GeneratedCode/Quantities/Acceleration.g.cs [hunk 1]
```diff
@@ -736,9 +736,10 @@ public static string GetAbbreviation(AccelerationUnit unit)
 
         public int CompareTo(object obj)
         {
-            if (obj == null) throw new ArgumentNullException("obj");
-            if (!(obj is Acceleration)) throw new ArgumentException("Expected type Acceleration.", "obj");
-            return CompareTo((Acceleration) obj);
+            if(obj is null) throw new ArgumentNullException("obj");
+            if(!(obj is Acceleration)) throw new ArgumentException("Expected type Acceleration.", "obj");
```
Comment (angularsen) [inline UnitsNet/GeneratedCode/Quantities/Acceleration.g.cs]: Let's add `nameof(obj)` too.

File: UnitsNet/GeneratedCode/Quantities/Acceleration.g.cs [hunk 2]
```diff
@@ -736,9 +736,10 @@ public static string GetAbbreviation(AccelerationUnit unit)
 
         public int CompareTo(object obj)
         {
-            if (obj == null) throw new ArgumentNullException("obj");
-            if (!(obj is Acceleration)) throw new ArgumentException("Expected type Acceleration.", "obj");
-            return CompareTo((Acceleration) obj);
+            if(obj is null) throw new ArgumentNullException("obj");
```
Comment (angularsen) [inline UnitsNet/GeneratedCode/Quantities/Acceleration.g.cs]: Didn't know you could do `is null`, now I know!

File: UnitsNet/GeneratedCode/Quantities/Acceleration.g.cs (hunk 1 above)
Comment (tmilnthorp) [reply UnitsNet/GeneratedCode/Quantities/Acceleration.g.cs]: Didn't know you could do this either 👍 

Done!

**Comment by angularsen** [comment]
You have to sync again, just merged a PR.

**Comment by tmilnthorp** [comment]
Done!

**Pull request merged**